### ISSUE TITLE
Linking to FAQ 'href' does not work within {% url %} tag

### ISSUE THREAD

**Issue by jennifer-shehane** [cypress-io/cypress-documentation#353]
The link below throws an error during the build that the id cannot be found:

```
{% url "Open Source" general-questions-faq#Is-Cypress-free-and-open-source %} 
```

If you write the link like below (thinking, maybe the icon is needed) it does not throw an error, but also the link does not go to the correct href

```
{% url "Open Source" general-questions-faq#fa-fa-angle-right-Is-Cypress-free-and-open-source %} 
```

**Comment by richdouglasevans** [comment]
Hi @jennifer-shehane 

I looked into this because I wanted to link to something in the FAQ and encountered this very issue.

The root cause is a sequencing one. The build does some eager validation of links to ensure that they are valid: in the case of validating _local_ links -- i.e. links to the Cypress documentation itself -- the validation check eagerly renders the source markdown and tries to find the link in the rendered HTML.

Unfortunately, this "simple" rendering of the source markdown doesn't account for the fact that the actual final rendered Cypress documentation will itself be transformed during the build process: so, it's looking at an unprocessed version of the HTML _but not the final HTML that's ultimately generated._ One of the helper tags -- `icon` -- transforms the HTML and this helper tag is not fired when the "simple" rendering happens.

As such, the validation fails. If the validation happened right at the end of the process using the final output of the build process then the validation would pass. (Assuming nothing was actually invalid.)

IMO the validation is happening at the wrong time. It's cool to fail fast but the validation should not be happening at the same time that the HTML is being generated 'cos that HTML may still be (and is) in the process of being transformed (mutated). Better to defer validation until the final HTML has been generated (and is immutable).

Since this is documentation about Cypress, I played around with doing the validation of the links using Cypress itself. It's pretty easy to do but the validation process does significantly increase the runtime of the Cypress test suite for the documentation. This is balanced by the fact that the build process is significantly faster because there's no validation happening at that time: the inline validation process means rendering Markdown to HTML repeatedly so being able to skip that is nice when you're repeatedly iterating on the documentation as I have been doing lately.

Certainly I could submit a PR to move the validation to the Cypress test suite and remove the validation during the build. There's no loss in the quality of the Cypress documentation 'cos links would still be being checked for validity (assuming one runs the Cypress test suite). 

Thoughts?

**Comment by jennifer-shehane** [comment]
Yes, we did initially try to run link validation through Cypress - and some tests on the table of contents links: https://github.com/cypress-io/cypress-documentation/blob/develop/cypress/integration/api_spec.coffee#L101 - that were at the time not performant, but since that issue has been closed and we have made improvements for Cypress - this test in particular works better now. General link validation may work better now also.

*The main objectives of link validation are:*
1. To not have dead links on the published documentation
2. If there is a dead or invalid link written/committed - to be able to easily locate and fix it. 

As long as objective #2 is met, I would be fine with ANY other link validation. No one especially enjoys the current implementation - but it is handy to get the errors as soon as you write an invalid link - as it seems easier to track down while you're writing. But, as you mentioned with links to headings containing icons - there are drawbacks to how it currently works. 

So, yes, PRs are welcome.
